python/ml4ir/applications/ranking/tests/test_rank_match_failure.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import unittest
+
+import numpy as np
+import tensorflow as tf
+import yaml
+from ml4ir.applications.ranking.model.metrics.aux_metrics_impl import RankMatchFailure
+from ml4ir.base.features.feature_config import FeatureConfig
+
+
+class RankMachFailureTest(tf.test.TestCase):
```

Comment:
@mohazahran This is ported from the old test @arvindsrikantan had added. Notice that I have only modified the calls and not the values. It's a good sanity check for parity.

Reference - https://github.com/salesforce/ml4ir/blob/6ff9238f5c7096530bba883ebd85a21e240d691e/python/ml4ir/applications/ranking/tests/test_rank_match_failure.py